ts/private/ts_project_worker.js

```diff
@@ -704,7 +704,7 @@ function createProgram(args, inputs, output, exit) {
     }
 
     function applySyntheticOutPaths() {
-        host.optionsToExtend.outDir = `${host.optionsToExtend.outDir}/${SYNTHETIC_OUTDIR}`;
+        host.optionsToExtend.outDir = `${host.optionsToExtend.outDir || host.optionsToExtend.rootDir || '.'}/${SYNTHETIC_OUTDIR}`;
```

Comment:
Why use `rootDir`? That seems wrong imo... 🤔

Reply:
Or is this `rootDir` different then `compilerOptions.rootDir`?

Reply:
Yes, `outDir || BUILD-dir` would be ideal. It's just that the long-lived worker process can only infer the BUILD directory from the work request. Right now this doesn't contain the BUILD directory. I just chose root rootDir since it happens to match BUILD-dir by default.

I can add a test, but more importantly: should we add the current package (aka `BUILD-dir`) to the work request explicitly instead of misusing rootDir?

Reply:
Note that I don't know this worker code well and I'm going off assumptions here. So maybe this `SYNTHETIC_OUTDIR` logic is different then what I'm assuming.

Can we simply avoid this `SYNTHETIC_OUTDIR` thing when `outDir` is not configured?